bin/tests/integration/named_metrics_tests.rs

```diff
@@ -5,44 +5,52 @@
 // https://opensource.org/licenses/MIT>, at your option. This file may not be
 // copied, modified, or distributed except according to those terms.
 
-use std::collections::HashMap;
 use std::net::*;
-use std::thread;
+use std::str::FromStr;
 use std::time::Duration;
 use test_support::subscribe;
 
+use hickory_client::client::{Client, ClientHandle};
+use hickory_proto::rr::rdata::A;
+use hickory_proto::rr::{DNSClass, Name, RData, RecordType};
+use hickory_proto::runtime::TokioRuntimeProvider;
+use hickory_proto::tcp::TcpClientStream;
+use hickory_proto::xfer::Protocol;
 use prometheus_parse::{Scrape, Value};
 use tokio::runtime::Runtime;
 
 use crate::server_harness::{ServerProtocol, SocketPorts, named_test_harness};
 
+const STORE_FILE_SUCCESS: [(&str, &str); 2] = [("store", "file"), ("success", "true")];
+const STORE_FILE_FAILED: [(&str, &str); 2] = [("store", "file"), ("success", "false")];
+const STORE_FORWARDER_SUCCESS: [(&str, &str); 2] = [("store", "forwarder"), ("success", "true")];
+const STORE_FORWARDER_FAILED: [(&str, &str); 2] = [("store", "forwarder"), ("success", "false")];
+
 #[test]
 fn test_prometheus_endpoint_startup() {
     subscribe();
 
     named_test_harness("example_forwarder.toml", |socket_ports| {
         let io_loop = Runtime::new().unwrap();
-
-        let metrics = &io_loop.block_on(fetch_parse_check_metrics(socket_ports));
-        let none = HashMap::new();
+        let metrics = &io_loop.block_on(fetch_parse_check_metrics(&socket_ports));
 
         // check process metrics
-        verify_metric(metrics, "process_cpu_seconds_total", &none, None);
-        verify_metric(metrics, "process_max_fds", &none, None);
-        verify_metric(metrics, "process_open_fds", &none, None);
-        verify_metric(metrics, "process_resident_memory_bytes", &none, None);
-        verify_metric(metrics, "process_start_time_seconds", &none, None);
-        verify_metric(metrics, "process_virtual_memory_bytes", &none, None);
+        verify_metric(metrics, "process_cpu_seconds_total", &[], None);
```

Comment:
Isolate the refactoring here in a separate commit from the metrics additions.

Reply:
Updated the commits, the re-factoring is now part of the initial commit.